contour/templates/contour-deployment.yaml

```diff
@@ -0,0 +1,79 @@
+{{ $root := . }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ template "contour.fullname" $root }}
+  labels:
+    app: {{ template "contour.fullname" $root }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+    heritage: {{ .Release.Service | quote }}
+    release: {{ .Release.Name | quote }}
+spec:
+  replicas: {{ .Values.deployment.contour.replicas }}
+  selector:
+    matchLabels:
+       app: {{ include "contour.fullname" $root }}
+       release: {{ .Release.Name }}
+  {{- with .Values.deployment.contour.strategy }}
+  strategy:
+{{ toYaml . | indent 4 }}
+  {{- end }}
+  template:
+    metadata:
+      labels:
+        app: {{ template "contour.fullname" $root }}
+        release: {{ .Release.Name }}
+      name: {{ template "contour.fullname" $root }}
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/port: "{{ .Values.deployment.contour.port.healthz }}"
+      {{- with .Values.deployment.contour.podAnnotations }}
+{{ toYaml . | indent 8 }}
+      {{- end }}
+    spec:
+      {{- with .Values.deployment.contour.imagePullSecrets }}
+      imagePullSecrets:
+{{ toYaml . | indent 8 }}
+      {{- end }}
+      containers:
+      - name: contour
+        image: "{{ .Values.deployment.contour.image.repository }}:{{ .Values.deployment.contour.image.tag }}"
+        imagePullPolicy: "{{ .Values.deployment.contour.image.imagePullPolicy }}"
+        command:
+        - "contour"
+        args:
+        - "serve"
+        - "--incluster"
+        - "--xds-address={{ .Values.deployment.contour.xdsAddress }}"
+        - "--xds-port={{ .Values.deployment.contour.port.xds }}"
+        - "--envoy-service-http-port={{ .Values.service.envoy.port.http }}"
+        - "--envoy-service-https-port={{ .Values.service.envoy.port.https }}"
+        {{- with .Values.deployment.contour.extraArgs }}
+{{ toYaml . | indent 8 }}
+        {{- end }}
+        resources:
+{{ toYaml .Values.deployment.contour.resources | indent 12 }}
+        ports:
+        - containerPort: {{ .Values.deployment.contour.port.xds }}
+          name: xds
+          protocol: TCP
+        - containerPort: {{ .Values.deployment.contour.port.healthz }}
+          name: debug
+          protocol: TCP
+        livenessProbe:
+          httpGet:
+            path: /healthz
+            port: {{ .Values.deployment.contour.port.healthz }}
+        readinessProbe:
+          httpGet:
+            path: /healthz
+            port: {{ .Values.deployment.contour.port.healthz }}
+      {{- with .Values.deployment.contour.affinity }}
+      affinity:
+{{ tpl (toYaml .) $root | indent 8 }}
+      {{- end }}
+      tolerations:
+      {{- with .Values.deployment.contour.tolerations }}
+{{ tpl (toYaml .) | indent 8 }}
+      {{- end }}
+      serviceAccountName: {{ if .Values.rbac.create }}{{ template "contour.fullname" . }}{{ else }}"{{ .Values.rbac.serviceAccountName }}"{{ end }}
```

Comment:
It appears to me that when `rbac` is disabled, it specifies `rbac.serviceAccountName`.